glean-core/csharp/Glean/GleanParser.cs

```diff
@@ -0,0 +1,208 @@
+
+        // This script runs a given Python module as a "main" module, like
+        // `python -m module`. However, it first checks that the installed
+        // package is at the desired version, and if not, upgrades it using `pip`.
+        //
+        // ** IMPORTANT**
+        // Keep this script in sync with the one in `glean-gradle-plugin/GleanGradlePlugin.groovy`.
+        private const string RunPythonScript = @"
+import importlib
+import subprocess
+import sys
+module_name = sys.argv[1]
+expected_version = sys.argv[2]
+try:
+    module = importlib.import_module(module_name)
+except ImportError:
+    found_version = None
+else:
+    found_version = getattr(module, '__version__')
+if found_version != expected_version:
+    subprocess.check_call([
+        sys.executable,
+        '-m',
+        'pip',
+        'install',
+        '--upgrade',
+        f'{module_name}=={expected_version}'
+    ])
+try:
+    subprocess.check_call([
+        sys.executable,
+        '-m',
+        module_name
+    ] + sys.argv[3:])
+except:
+    # We don't need to show a traceback in this helper script.
+    # Only the output of the subprocess is interesting.
+    sys.exit(1)
+";
+
+        /// <summary>
+        /// The Glean registry files to process.
+        /// </summary>
+        [Required]
+        public ITaskItem[] RegistryFiles { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        /// <summary>
+        /// The namespace in which to generate the Glean metrics.
+        /// </summary>
+        [Required]
+        public string Namespace { get; set; }
+
+        /// <summary>
+        /// This *MUST* only be used by the Glean SDK.
+        ///
+        /// Whether or not to allow using Glean reserved names. Defaults to
+        /// `false`.
+        /// </summary>
+        public bool AllowReserved { get; set; } = false;
+
+        // Make sure that we can see the output from the glean_parser when
+        // building.
+        protected override MessageImportance StandardOutputLoggingImportance
+        {
+            get { return MessageImportance.Normal; }
+        }
+
+        protected override string ToolName
+        { 
+            // TODO: it's python.exe on Win, python on linux/mac
+            get { return "python.exe"; }
+        }
+        
+        /// <summary>
+        /// Called to get the full path of the tool.
+        /// 
+        /// When returning `ToolExe`, this will look for the tool in the PATH.
+        /// If `ToolPath` is set, then the value of this function is ignored.
+        /// </summary>
+        /// <returns>A string representing the path to the tool</returns>
+        protected override string GenerateFullPathToTool()
+        { 
+            return ToolExe;
+        }
+
+        private string GetVirtualEnvironmentPath(string venvDir)
+        {
+            // TODO: "Scripts" is only there on windows. On Linux/Mac is "bin"
```

Comment:
Is there a bug to implement this?  I know we don't have any consumers but it might be good to track this TODO.  In the case of OS detection, I wonder if [something like this](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.isosplatform?view=netframework-4.7.2) would help.

Reply:
I bet we'll want to support Linux at least to access cheaper CI...

Reply:
Oh no bug, that's what I meant in the PR comment with "TODO:  Fix using the task on Windows vs Linux". I took care of it this morning.